src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceApiServiceSpec.scala

```diff
@@ -243,6 +243,52 @@ class WorkspaceApiServiceSpec extends FlatSpec with HttpService with ScalatestRo
       }
   }
 
+  it should "return 400 when batch upserting an entity with invalid update operations" in withTestDataApiServices { services =>
+    Post(s"/workspaces/${testData.workspace.namespace}/${testData.workspace.name}/entities/batchUpsert", HttpEntity(ContentTypes.`application/json`, Seq(EntityUpdateDefinition("s1", "samples", Seq(RemoveListMember("bingo", AttributeString("a")))), EntityUpdateDefinition("s2", "samples", Seq(AddUpdateAttribute("newAttribute", AttributeString("foo")))), EntityUpdateDefinition("s3", "samples", Seq(RemoveListMember("bingo", AttributeString("a"))))).toJson.toString())) ~>
```

Comment:
Could you move these updates to a  `val` with one update per line, for readability?